.claude/hooks/check-branch.sh

```
@@ -1,13 +1,47 @@
-    echo "❌ BLOCKED: On main branch. Create a feature branch first:"
-    echo "   git checkout -b <type>/<description>"
+# Only care about main branch
+[[ "$BRANCH" != "main" ]] && exit 0
+
+# Read the tool input to get the command
+INPUT=$(cat)
+COMMAND=$(echo "$INPUT" | python3 -c "import sys,json; print(json.load(sys.stdin).get('command',''))" 2>/dev/null)
+
+# Reject commands with shell metacharacters that could chain commands
+# This prevents bypasses like: git status && rm -rf /
+if echo "$COMMAND" | grep -qE '[;|&`$()]|>\s*[^&]'; then
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)